sky/backends/backend_utils.py

```diff
@@ -481,226 +481,83 @@ def add_cluster(
-                ip = external_worker_ips[idx]
-                if docker_user is not None:
-                    docker_proxy_command = docker_proxy_command_generator(ip)
-                    ip = 'localhost'
-                # TODO(romilb): Update port number when k8s supports multinode
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, host_name, ip, username, key_path,
-                    proxy_command, port, docker_proxy_command)
-
-        # All workers go to SKY_USER_FILE_PATH/ssh/{cluster_name}
-        for i, line in enumerate(extra_config):
-            if line.strip() in host_lines:
-                idx = host_lines.index(line.strip())
-                prev_line = extra_config[i - 1] if i > 0 else ''
-                if prev_line.strip().startswith(sky_autogen_comment):
-                    host_name = worker_names[idx]
-                    overwrites[idx] = True
-                    overwrite_begin_idxs[idx] = i - 1
-                ip = external_worker_ips[idx]
-                if docker_user is not None:
-                    docker_proxy_command = docker_proxy_command_generator(ip)
-                    ip = 'localhost'
-                # TODO(romilb): Update port number when k8s supports multinode
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, host_name, ip, username, key_path,
-                    proxy_command, port, docker_proxy_command)
-
-        # This checks if all codegens have been created.
-        for idx, ip in enumerate(external_worker_ips):
-            if docker_user is not None:
-                docker_proxy_command = docker_proxy_command_generator(ip)
-                ip = 'localhost'
-            if not codegens[idx]:
-                codegens[idx] = cls._get_generated_config(
-                    sky_autogen_comment, worker_names[idx], ip, username,
-                    key_path, proxy_command, port, docker_proxy_command)
-
-        for idx in range(len(external_worker_ips)):
-            # Add (or overwrite) the new config.
-            overwrite = overwrites[idx]
-            overwrite_begin_idx = overwrite_begin_idxs[idx]
-            codegen = codegens[idx]
-            assert codegen is not None, (codegens, idx)
-            if overwrite:
-                assert overwrite_begin_idx is not None
-                updated_lines = codegen.splitlines(keepends=True) + ['\n']
-                extra_config[overwrite_begin_idx:overwrite_begin_idx +
-                             len(updated_lines)] = updated_lines
-                with open(extra_config_path, 'w') as f:
-                    f.write(''.join(extra_config).strip())
-                    f.write('\n' * 2)
-            else:
-                with open(extra_config_path, 'a') as f:
-                    f.write(codegen)
-                    f.write('\n')
+                ['-W', '%h:%p', f'{auth_config["ssh_user"]}@{ips[0]}'])
```

Comment:
Why do we changed it the `ips[0]` instead of the original generator? Will this cause the docker container on the workers failed to be connected?
Please check the following lines:
https://github.com/skypilot-org/skypilot/blob/7609e229a448a8f7d27642a8d34d33b6e0534702/sky/backends/backend_utils.py#L623-L626

Reply:
Aah i see yeah! For some reason the code has refactored to that. Will fix!